ArduCopter/AP_Arming.cpp

```diff
@@ -12,24 +12,18 @@ void AP_Arming_Copter::update(void)
         pre_arm_display_counter = 0;
     }
 
-    if (pre_arm_checks(display_fail)) {
-        set_pre_arm_check(true);
-    }
+    set_pre_arm_check(pre_arm_checks(display_fail));
 }
 
 // performs pre-arm checks and arming checks
 bool AP_Arming_Copter::all_checks_passing(bool arming_from_gcs)
 {
-    if (pre_arm_checks(true)) {
-        set_pre_arm_check(true);
-    } else {
-        return false;
-    }
+    set_pre_arm_check(pre_arm_checks(true));
```

Comment:
Is this backwards?

Reply:
Sadly, no, I don't think so.

set_pre_arm_checks sets some variables based on the result of pre_arm_checks.

The argument to pre_arm_checks is whether to be noisy or not.